app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,37 @@
+class MessagesController < ApplicationController
+  before_action do
+   @conversation = Conversation.find(params[:conversation_id])
+  end
+
+  def index
+   @messages = @conversation.messages
+    if @messages.length > 10
+     @over_ten = true
+     @messages = @messages[-10..-1]
+    end
+    if params[:m]
+     @over_ten = false
+     @messages = @conversation.messages
+    end
+   if @messages.last
+    if @messages.last.user_id != current_user.id
+     @messages.last.read = true;
+    end
+   end
+  @message = @conversation.messages.new
+   end
+
+  def new
+   @message = @conversation.messages.new
+  end
+
+  def create
+    @message = @conversation.messages.new(message_params)
+    @message.save
```

Comment:
what if it fails?

If that can't happen (or if we don't care), you can use create in the line above